### PR TITLE
Aggregate Databricks 11.3 shim in the nightly dist jar

### PR DESCRIPTION
**Pull request by NvTimLiu**
part of https://github.com/NVIDIA/spark-rapids/issues/7389

`DIST_PROFILE_OPT` to aggregate db330 shim into dist jar

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by pxLi** [comment]
Convert to draft to avoid accidental merge, this must be merged along w/ internal CI update

File: jenkins/spark-nightly-build.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ DIST_POM_FPATH="$DIST_PL/target/extra-resources/META-INF/maven/$ART_GROUP_ID/$AR
 DIST_PROFILE_OPT=-Dincluded_buildvers=$(IFS=,; echo "${SPARK_SHIM_VERSIONS[*]}")
 DIST_INCLUDES_DATABRICKS=${DIST_INCLUDES_DATABRICKS:-"true"}
 if [[ "$DIST_INCLUDES_DATABRICKS" == "true" ]]; then
-    DIST_PROFILE_OPT="$DIST_PROFILE_OPT,312db,321db"
+    DIST_PROFILE_OPT="$DIST_PROFILE_OPT,312db,321db,330db"
```
Comment (pxLi) [inline jenkins/spark-nightly-build.sh:44]: nit: could generate the list from pom file as our single source in version-def script

Comment (NvTimLiu) [reply jenkins/spark-nightly-build.sh]: Good point
updated

**Comment by NvTimLiu** [comment]
BTW, there should be some doc and source update support DB330 from dev team, e.g.

- https://github.com/NVIDIA/spark-rapids/blob/branch-23.02/docs/additional-functionality/rapids-shuffle.md
```
<html>
<body>
<!--StartFragment-->

Databricks 11.3 | com.nvidia.spark.rapids.spark330db.RapidsShuffleManager
-- | --

<!--EndFragment-->
</body>
</html>
```

- Missing  source file : `spark330db.RapidsShuffleManager` under the path `sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/spark330db/` ?

**Comment by NvTimLiu** [comment]
build

**Comment by pxLi** [comment]
cc @tgravescs @gerashegalov @nartal1 to help review this update, thx~

**Comment by NvTimLiu** [comment]
> BTW, there should be some doc and source update support DB330 from dev team, e.g.
> 
> * Missing 'spark330db.RapidsShuffleManager' in https://github.com/NVIDIA/spark-rapids/blob/branch-23.02/docs/additional-functionality/rapids-shuffle.md ?
> 
> ```
> <html>
> <body>
> 
> 
> Databricks 11.3 | com.nvidia.spark.rapids.spark330db.RapidsShuffleManager
> -- | --
> 
> 
> </body>
> </html>
> ```
> 
> * Missing  source file : `spark330db.RapidsShuffleManager` under the path `sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/spark330db/` ?


Unresolved, track with the issue https://github.com/NVIDIA/spark-rapids/issues/7389

File: dist/pom.xml [hunk 2]
```diff
@@ -133,6 +134,14 @@
...
             </properties>
         </profile>
+        <profile>
+            <id>databricks</id>
+            <properties>
+                <included_buildvers>
+                    ${databricks.buildvers}
+                </included_buildvers>
+            </properties>
+        </profile>
```
Comment (gerashegalov) [inline dist/pom.xml:144]: how are we building `dist` on CI?  Does it mean we now publish a databricks-only artifact?  If it's just to get the list of databricks versions, is it worth creating a profile if we can just use `-Dexpression=databricks.buildvers` to do this

Comment (NvTimLiu) [reply dist/pom.xml:144]: @gerashegalov  `databricks` profile is to get databricks shims aggregating into the dist jar for the Nightly CI, it does NOT mean we publish databricks-only artifact. 

We define this `databricks` profile to better reuse our CI script to get DB shims, https://github.com/NVIDIA/spark-rapids/blob/branch-23.02/jenkins/version-def.sh#L54. 

 Yes we can just use `-Dexpression=databricks.buildvers` to do this, and if you think introducing a new profile is confusing, we can follow it up.

Comment (gerashegalov) [reply dist/pom.xml:144]: ok let us leave it like this for consistency.  we can refactor all these profiles independently of the Databricks 11.3 work

**Pull request merged**